src/test/java/com/google/cloud/tools/managedcloudsdk/install/DownloaderTest.java

```diff
@@ -175,8 +175,8 @@ public void run() {
                   Assert.fail("InterruptedException expected but not thrown.");
                 } catch (InterruptedException ex) {
                   Assert.assertEquals("Download was interrupted", ex.getMessage());
-                } catch (IOException e) {
-                  Assert.fail("Test failed due to IOException");
+                } catch (IOException ex) {
+                  throw new AssertionError("Test failed due to IOException", ex);
```

Comment:
I actually see a bigger problem in this test. Failing in a different thread won't fail the actual test, AFAIK. We need to propagate the result to the main test thread instead. It'll be a bit involved (not so much). Let me file an issue and tackle this myself.

Reply:
Yeah, it appears like I missed something with this test. The fix is to use an executor service and Callable, and the exception will be propagated.

Reply:
Ah, good to know. I was trying a different approach, but the executor services seems nice.